src/ui/common/src/components/pages/workflow/id/index.tsx

```diff
@@ -116,11 +123,25 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
     }
   }, [workflow.selectedResult, urlSearchParams, navigate]);
 
+  // EFFECT 3: Load workflow metadata.
+  // This useEffect is effectively only called on component mount. It loads
+  // the base workflow metadata as well as metadata about any integrations
+  // in order to populate the UI.
   useEffect(() => {
     dispatch(handleGetWorkflow({ apiKey: user.apiKey, workflowId }));
     dispatch(handleLoadIntegrations({ apiKey: user.apiKey }));
   }, [dispatch, user.apiKey, workflowId]);
 
+  // EFFECT 4: Gather selected workflow index.
+  // When the workflow Redux store's DAG results are populated or when we navigate
+  // to a different version, we iterate through the full list of results and set the
+  // index in both Redux and in our local state.
+  // NOTE(vikram): There are two annoying bits of tech debt in this code:
+  // 1. It's not clear that this needs to be a different from Effect 2 the one where we
+  // navigate to a different search param. They seem to be focused on the same bits of
+  // functionality. (See ENG-2569.)
+  // 2. Less critical, but it's annoying that we have to track selectedResultIdx in local
```

Comment:
re: 2 This started out as a state variable because it didn't need to be shared anywhere else. Where else is this going to be needed outside of here?